src/Thumbnail/GeneratableThumbnailInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\MediaBundle\Thumbnail;
+
+use Sonata\MediaBundle\Model\MediaInterface;
+use Sonata\MediaBundle\Provider\MediaProviderInterface;
+
+/**
+ * @author Jordi Sala Morales <jordism91@gmail.com>
+ */
+interface GeneratableThumbnailInterface
```

Comment:
I don't know how correct it can be, but I think "Generable" fits better than "Generatable" in this case.
BTW, with a too specific name like the one this interface currently has, the `delete()` method seems unrelated to its responsibilities. IMO, the case could be different if the name were `Thumbnail(Manager|Handler)Interface` or similar.

Reply:
Renamed to Generable. I kinda don't like super generics names because we will end up with more methods there instead of more interfaces.

I think about it this way: Thumbnails that are generated should be also deleted at some point (when you remove the media). There are the File Thumbnails, that are for files and can't be generated, so they cannot be removed either. And the last case if for Liip thumbnails, generated on demand and stored on some cache folder (no need to delete them because they will be removed on cache removal).

wdyt?

Reply:
It's fine for me. I just want to point out that this level of specificity might require a `DeletableThumbnailInterface`.
But I don't want to block this PR.